### PR TITLE
added numpy broadcast class and its tests

### LINKED ISSUE #19278: broad

### PR DESCRIPTION
**Pull request by ShreyanshBardia**
closes #19278

### CODE REVIEW

File: ivy/functional/frontends/numpy/broadcast/methods.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# local
+import ivy
+from ivy.functional.frontends.numpy.func_wrapper import to_ivy_arrays_and_back
+
+
+class broadcast:
+    @to_ivy_arrays_and_back
+    def __init__(self, *args):
+        data = ivy.broadcast_arrays(*map(ivy.array, args))
+        self.__shape = data[0].shape
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/broadcast/methods.py:10]: Can we use one underscore here 
something like `self._shape` as this one doesn't look good

**Review by hirwa-nshuti — COMMENTED** [review]
Hey @ShreyanshBardia

Thanks for the PR 🙂
Most of the changes looks good to be merged only a few changes are needed here😊
Thanks

File: ivy/functional/frontends/numpy/broadcast/methods.py (hunk 1 above)
Comment (ShreyanshBardia) [reply ivy/functional/frontends/numpy/broadcast/methods.py]: Hi @hirwa-nshuti, updated it.

**Review by hirwa-nshuti — APPROVED** [review]
Hey @ShreyanshBardia 

Thanks for the PR all looks good to me😊
Going to merge🙂

**Pull request merged**